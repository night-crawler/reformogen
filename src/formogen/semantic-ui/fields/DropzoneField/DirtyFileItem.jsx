import React from 'react';
import PropTypes from 'prop-types';
import { List, Image, Button } from 'semantic-ui-react';

import { splitExt, bytesToSize } from '../../../utils';
import { fileTypeImageMapping, UNKNOWN_FILE_TYPE } from '../../../fileTypeImageMapping';
import { CaptionTruncator } from '../../CaptionTruncator';

import { fileType } from './propTypes';


DirtyFileItem.propTypes = {
  file: fileType,
  getPreviewImage: PropTypes.func,
  onDelete: PropTypes.func,
};
DirtyFileItem.defaultProps = {
  getPreviewImage: file => 
    fileTypeImageMapping[ splitExt(file.name)[1] ] || UNKNOWN_FILE_TYPE,
  onDelete: () => // eslint-disable-next-line
    console.warn('DirtyFileItem.onDelete'),
};
export function DirtyFileItem(props) {
  const [ fileName, fileExt ] = splitExt(props.file.name);
  const sizeRepr = `[${bytesToSize(props.file.size)}]`;

  // TODO: watermelon come here, motherfucker!
  return (
    <List.Item>
      <Image 
        src={ props.getPreviewImage(props.file) } 
        verticalAlign='middle' size='mini'
      />

      <List.Content style={ { width: '100%' } }>
        <strong>{ sizeRepr }&nbsp;</strong>
        <CaptionTruncator caption={ fileName } width='50%' />
        <span>.{ fileExt }</span>
      </List.Content>

      <Button 
        icon='remove' size='mini' attached='right' floated='right' 
        onClick={ props.onDelete }
      />
    </List.Item>
  );
}
