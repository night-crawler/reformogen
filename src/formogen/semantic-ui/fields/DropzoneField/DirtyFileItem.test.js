import { Button } from 'semantic-ui-react';

import { DirtyFileItem } from './DirtyFileItem';

const file = {
  lastModified: 1,
  lastModifiedDate: {},
  name: 'sample.png',
  size: 1000,
  type: 'application/trash-x',
};


it('<DirtyFileItem />', () => {
  const onDeleteFn = jest.fn();
  const wrapper = shallow(
    <DirtyFileItem
      file={ file }
      onDelete={ onDeleteFn }
    />
  );
  expect(wrapper).toMatchSnapshot();
  
  wrapper.find(Button).simulate('click');
  expect(onDeleteFn).toHaveBeenCalled();
});
