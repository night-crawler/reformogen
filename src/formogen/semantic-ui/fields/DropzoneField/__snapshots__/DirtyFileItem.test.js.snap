// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DirtyFileItem /> 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DirtyFileItem
    file={
      Object {
        "lastModified": 1,
        "lastModifiedDate": Object {},
        "name": "sample.png",
        "size": 1000,
        "type": "application/trash-x",
      }
    }
    getPreviewImage={[Function]}
    onDelete={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Image
          as="img"
          size="mini"
          src="test-file-stub"
          ui={true}
          verticalAlign="middle"
        />,
        <ListContent
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <strong>
            [1000 Bytes]
             
          </strong>
          <CaptionTruncator
            caption="sample"
            width="50%"
          />
          <span>
            .
            png
          </span>
        </ListContent>,
        <Button
          as="a"
          attached="right"
          floated="right"
          icon="remove"
          onClick={[MockFunction]}
          role="button"
          size="mini"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "as": "img",
          "size": "mini",
          "src": "test-file-stub",
          "ui": true,
          "verticalAlign": "middle",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <strong>
              [1000 Bytes]
               
            </strong>,
            <CaptionTruncator
              caption="sample"
              width="50%"
            />,
            <span>
              .
              png
            </span>,
          ],
          "style": Object {
            "width": "100%",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "[1000 Bytes]",
                " ",
              ],
            },
            "ref": null,
            "rendered": Array [
              "[1000 Bytes]",
              " ",
            ],
            "type": "strong",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "caption": "sample",
              "width": "50%",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                ".",
                "png",
              ],
            },
            "ref": null,
            "rendered": Array [
              ".",
              "png",
            ],
            "type": "span",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "as": "a",
          "attached": "right",
          "floated": "right",
          "icon": "remove",
          "onClick": [MockFunction],
          "role": "button",
          "size": "mini",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Image
            as="img"
            size="mini"
            src="test-file-stub"
            ui={true}
            verticalAlign="middle"
          />,
          <ListContent
            style={
              Object {
                "width": "100%",
              }
            }
          >
            <strong>
              [1000 Bytes]
               
            </strong>
            <CaptionTruncator
              caption="sample"
              width="50%"
            />
            <span>
              .
              png
            </span>
          </ListContent>,
          <Button
            as="a"
            attached="right"
            floated="right"
            icon="remove"
            onClick={[MockFunction]}
            role="button"
            size="mini"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "img",
            "size": "mini",
            "src": "test-file-stub",
            "ui": true,
            "verticalAlign": "middle",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <strong>
                [1000 Bytes]
                 
              </strong>,
              <CaptionTruncator
                caption="sample"
                width="50%"
              />,
              <span>
                .
                png
              </span>,
            ],
            "style": Object {
              "width": "100%",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "[1000 Bytes]",
                  " ",
                ],
              },
              "ref": null,
              "rendered": Array [
                "[1000 Bytes]",
                " ",
              ],
              "type": "strong",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "caption": "sample",
                "width": "50%",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  ".",
                  "png",
                ],
              },
              "ref": null,
              "rendered": Array [
                ".",
                "png",
              ],
              "type": "span",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "as": "a",
            "attached": "right",
            "floated": "right",
            "icon": "remove",
            "onClick": [MockFunction],
            "role": "button",
            "size": "mini",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
